resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -412,17 +412,22 @@ private void finishResolve() {
         final int tries = maxAllowedQueries - allowedQueries;
         final StringBuilder buf = new StringBuilder(64);
 
-        buf.append("failed to resolve ");
+        buf.append("failed to resolve '");
         buf.append(hostname);
+        buf.append('\'');
 
         if (tries > 1) {
-            buf.append(" after ");
-            buf.append(tries);
+            if (tries < maxAllowedQueries) {
+                buf.append(" after ");
+                buf.append(tries);
+                buf.append(" queries");
+            } else {
+                buf.append(". Exceeded max queries per resolve ").append(maxAllowedQueries);
```

Comment:
consider adding a space on the end of this line. then you could share the `trace` appending code and get rid of some conditionals